packages/ui-inputs/src/search-input.tsx

```diff
@@ -0,0 +1,146 @@
+import * as React from 'react'
+import { useEffect, useState } from 'react'
+import { InputElement, InputRef, TextInput } from './text-input';
+import { Plus24Icon, Search24Icon } from '@kaltura-path/ui-icons';
+import { Theme } from './theme/theme';
+import { createUseStyles, theming } from './theme';
+
+const classNames = require('classnames');
+
+export interface SearchInputFieldProps {
+    /**
+     * Initial value for controlled input
+     * @default undefined
+     * */
+    value?: string;
+    /**
+     * Initial value for uncontrolled input
+     * @default undefined
+     * */
+    defaultValue?: string;
+    /**
+     * Set input in disabled state which prevents user input and changes input style
+     * @default undefined
+     * */
+    disabled?: boolean;
+    /**
+     * Placeholder text which is displayed when no initial value or user input provided
+     * @default undefined
+     * */
+    placeholder?: string
+    /**
+     * Ref provides a way to access DOM nodes or React elements created in the render method
+     * @default undefined
+     * */
+    inputRef?: InputRef;
+    /**
+     * Flag that indicates if input has an error, changes input style
+     * @default false
+     * */
+    hasError?: boolean;
+    /**
+     * Sets input in busy state by displaying animation indicating busy state which replace the search icon
+     * @default false
+     * */
+    isBusy?: boolean;
+    /**
+     * Event callback which is triggered after a user have typed something into an input field
+     * @default undefined
+     * */
+    onChange?: (event: React.ChangeEvent<InputElement>) => void;
```

Comment:
I'm thinking out loud. Not sure if we want to provide the underline InputElement and not the actual DOM input element. what will the user need to do in order to get the value using a ref?
If you think like this as well provide the actual DOM input. if not just let me know what you think about it.

Reply:
It is an input element, I've just added an alias for it, so we don't need to write "or null" each time in the typings
`export type InputElement = HTMLInputElement | null;`